src/main/java/uk/gov/pay/api/model/search/directdebit/DirectDebitSearchMandatesParams.java

```diff
@@ -0,0 +1,175 @@
+package uk.gov.pay.api.model.search.directdebit;
+
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.QueryParam;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class DirectDebitSearchMandatesParams {
+
+    @QueryParam("reference")
+    private String reference;
+
+    @QueryParam("state")
+    private String state;
+
+    @QueryParam("bank_statement_reference")
+    private String bankStatementReference;
+
+    @QueryParam("email")
+    private String email;
+
+    @QueryParam("name")
+    private String name;
+
+    @QueryParam("from_date")
+    private ZonedDateTime fromDate;
+
+    @QueryParam("to_date")
+    private ZonedDateTime toDate;
+
+    @DefaultValue("1")
+    @QueryParam("page")
+    private int page;
+
+    @DefaultValue("500")
+    @QueryParam("display_size")
+    private int displaySize;
+
+    private DirectDebitSearchMandatesParams(DirectDebitSearchMandatesParamsBuilder builder) {
+        this.reference = builder.reference;
+        this.state = builder.state;
+        this.bankStatementReference = builder.bankStatementReference;
+        this.email = builder.email;
+        this.name = builder.name;
+        this.fromDate = builder.fromDate;
+        this.toDate = builder.toDate;
+        this.page = builder.page;
+        this.displaySize = builder.displaySize;
+    }
+
+    public Optional<String> getReference() {
+        return Optional.ofNullable(reference);
+    }
+
+    public Optional<String> getState() {
+        return Optional.ofNullable(state);
+    }
+
+    public Optional<String> getBankStatementReference() {
+        return Optional.ofNullable(bankStatementReference);
+    }
+
+    public Optional<String> getEmail() {
+        return Optional.ofNullable(email);
+    }
+
+    public Optional<String> getName() {
+        return Optional.ofNullable(name);
+    }
+
+    public Optional<ZonedDateTime> getFromDate() {
+        return Optional.ofNullable(fromDate);
+    }
+
+    public Optional<ZonedDateTime> getToDate() {
+        return Optional.ofNullable(toDate);
+    }
+
+    public int getPage() {
+        return page;
+    }
+
+    public int getDisplaySize() {
+        return displaySize;
+    }
+
+    public Map<String, String> paramsAsMap() {
+        Map<String, String> returnableMap = new HashMap<>();
```

Comment:
I’d just call this `params`, since that’s what it is.

You could also declare this as `var params = new HashMap<String, String>();`.

Reply:
👍